sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java

```diff
@@ -18,38 +17,42 @@
 import java.io.Closeable;
 
 /**
- * CosmosClient with encryption support.
- * We have static method in this class which will takes two inputs
- * {@link CosmosClient} and {@link EncryptionKeyWrapProvider}  and creates cosmosEncryptionClient as shown below.
- * <pre>
- * {@code
- * CosmosEncryptionClient cosmosEncryptionClient =
- * CosmosEncryptionClient.createCosmosEncryptionClient(cosmosClient, encryptionKeyWrapProvider);
- * }
- * </pre>
+ * Provides a client-side logical representation of the Azure Cosmos DB service.
+ * Calls to CosmosClient API's are blocked for completion.
  */
 @ServiceClient(builder = CosmosEncryptionClientBuilder.class)
 public final class CosmosEncryptionClient implements Closeable {
     private final static Logger LOGGER = LoggerFactory.getLogger(CosmosEncryptionAsyncClient.class);
     private final CosmosEncryptionAsyncClient cosmosEncryptionAsyncClient;
-    private EncryptionKeyWrapProvider encryptionKeyWrapProvider;
+    private final KeyEncryptionKeyResolver keyEncryptionKeyResolver;
     private final CosmosAsyncClient cosmosAsyncClient;
     private final CosmosClient cosmosClient;
     private final CosmosClientAccessor cosmosClientAccessor;
+    private final String keyEncryptionKeyResolverName;
 
-    CosmosEncryptionClient(CosmosClient cosmosClient, EncryptionKeyWrapProvider encryptionKeyWrapProvider) {
+    CosmosEncryptionClient(CosmosClient cosmosClient, KeyEncryptionKeyResolver keyEncryptionKeyResolver,
+                           String keyEncryptionKeyResolverName) {
         this.cosmosClientAccessor = CosmosClientHelper.geCosmosClientAccessor();
-        this.encryptionKeyWrapProvider = encryptionKeyWrapProvider;
+        this.keyEncryptionKeyResolver = keyEncryptionKeyResolver;
         this.cosmosClient = cosmosClient;
         this.cosmosAsyncClient = this.cosmosClientAccessor.getCosmosAsyncClient(cosmosClient);
-        this.cosmosEncryptionAsyncClient = new CosmosEncryptionAsyncClient(cosmosAsyncClient, encryptionKeyWrapProvider);
+        this.cosmosEncryptionAsyncClient = new CosmosEncryptionAsyncClient(cosmosAsyncClient, keyEncryptionKeyResolver,
+            keyEncryptionKeyResolverName);
+        this.keyEncryptionKeyResolverName = keyEncryptionKeyResolverName;
     }
 
     /**
-     * @return the encryption key wrap provider
+     * @return the key encryption key resolver
      */
-    public EncryptionKeyWrapProvider getEncryptionKeyWrapProvider() {
-        return encryptionKeyWrapProvider;
+    public KeyEncryptionKeyResolver getEncryptionKeyWrapProvider() {
+        return this.keyEncryptionKeyResolver;
+    }
+
+    /**
+     * @return the key encryption key resolver name
+     */
+    public String getKeyEncryptionKeyResolverName() {
+        return keyEncryptionKeyResolverName;
```

Comment:
--> this.cosmosEncryptionAsyncClient.getKeyEncryptionKeyResolverName() ? 
and then we can remove the reference of keyEncryptionKeyResolverName  in this class?

Reply:
We want sync and async constructor same which take both keyEncryptionKeyResolverName and keyEncryptionKeyResolver, so its better if it rely on its own variable although value will be same